File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
     let burger = 0;
     let stack = [];
     for(const ingre of ingredient) {
         stack.push(ingre);
         if(stack[stack.length-4] === 1 
            && stack[stack.length-3] === 2 &&  
            stack[stack.length-2] === 3 && 
            stack[stack.length-1] === 1) {
                 stack.pop();
                 stack.pop();
                 stack.pop();
                 stack.pop();
                 burger++;
         } 
     }
     return burger;
}

File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int burger = 0;
        Stack<Integer> stack = new Stack<>();
        for(int ingre: ingredient) {
            stack.push(ingre);
            if(stack.size() >= 4) {
               if(stack.get(stack.size() - 1) == 1 &&
               stack.get(stack.size() - 2) == 3 &&
               stack.get(stack.size() - 3) == 2 &&
               stack.get(stack.size() - 4) == 1
              ) {
                for(int i = 0; i < 4; i++) {
                    stack.pop();
                }
                burger++;
            } 
            }
            
        }
        return burger;
    }
}